test/test_modules.py

```diff
@@ -16,11 +17,35 @@ class TestModule(TestCase):
     precision = 1e-5
     rel_tol = 1e-5
 
+    def _assert_module_parameters_and_buffer_are(self, module, device, dtype, device_id=None):
+        # Check device placement and dtype for created parameters and buffers.
+        # Only verify floating point dtypes since that's what the kwarg or methods
+        # such as `float()` applies to.
+        def _check_module(items, name):
+            for item_name, item in items:
+                self.assertEqual(
+                    str(item.device), device,
+                    f'{name} {item_name} is on {item.device.type} instead of the expected device {device}')
```

Comment:
nit: error message is not as precise as it could be:
```
Buffer weight is on cuda instead of the expected device cuda:0
```

Should probably use `item.get_device()` here. Does there need to be a separate device ID check since the ID is part of the device string?